src/vl-map-layer.js

```diff
@@ -113,13 +120,17 @@ export class VlMapLayer extends vlElement(HTMLElement) {
   }
 
   /**
-   * Zet de OpenLayers kaartlaag stijl.
+   * Voeg een kaartlaag stijl toe.
    *
-   * @param {ol.style} style
+   * @param {VlMapLayerStyle} style een kaartlaag stijl
    */
   set style(style) {
```

Comment:
Breaking change!

Best in de PR een beschrijving toevoegen en voor de afnemers een uitleg/voorbeeld om te migreren bij een upgrade. Dit kan dan toegevoegd worden aan de release.

Reply:
Deze change heeft als gevolg dat afnemers niet meer op een vl-map-layer een OpenLayers stijl kunnen definiëren, maar verplicht worden om de vl-map-layer-style te gebruiken. Als de vl-map-layer-style onvoldoende features blijkt te hebben om de huidige applicatie stijl te kunnen bepalen, kan de upgrade uitgesteld worden in afwachting van een uitbreiding.

Reply:
Normaal zou voor ons alles aanwezig moeten zijn, maar goeie vraag voor de andere afnemers uiteraard.

Reply:
Wij zetten programmatisch de stijl (a.d.h.v. een eigen stijl factory), dus als dat niet meer mogelijk is het breaking voor ons.

Ter info, wij hebben driehoek, ruit, cirkel en vierkant.

Reply:
setStyle en setVlStyle dan of zo ...?
maar kunt ge de 2 dan "mixen" als ge meerdere stijlen hebt?

Reply:
geen idee wat het gedrag dan is...

risico van deze manier is dat we voor andere zaken dan ook met 2 setters (en misschien getters) gaan zitten: vl en ol variant

of zou VlMapLayerStyle ook als wrapper kunnen gebruikt worden en daar de ol style zetten?

Reply:
Dat laatste begrijp ik niet goed @driesva, de VlMapLayerStyle IS een wrapper voor de ol style nu ...
Alleen verplichten we dan dus om alle style via VlMapLayerStyle te gaan doen (wat ik persoonlijk goed vind, "declarative first"), maar de vraag is dus of iedereen daar akkoord mee kan gaan of dat we toch nog een "achterpoortje" moeten voorzien.
Wat mij betreft, leve encapsulatie, en zo weinig mogelijk achterpoortjes ...

Reply:
Met wrapper bedoelde ik rechtstreeks de OL style zetten. Nu is er toch wrapping nadat deze eerst werd gebuild obv attributen?

Reply:
Ah ik ben mee, je bedoelt de ol style op de VlMapLayerStyle zetten, niet op de VlMapLayer ...
Dat zou nog een optie kunnen zijn vermoed ik ja.

Reply:
Even onderzocht, en op zich zou je inderdaad met 2 functies kunnen werken op vl-map-layer:
- addStyle: een VlMapLayerStyle toevoegen, die nodig is om declaratief de stijl(en) onder de laag te zetten
- set style: de oude behouden, waarbij rechtstreeks een OpenLayers stijl kan gezet worden, die zal wel de (eventuele) onderliggende VlMapLayerStyle's verwijderen dan; eventueel kan deze methode deprecated worden geannoteerd, in favor van de nieuwe methode, waarbij aangeraden wordt om declaratief de VlMapLayerStyle te gebruiken (moet dan inderdaad nog aangepast worden zodat deze de zaken toelaat dat nu nog niet mogelijk zijn)

Reply:
En de suggestie van @driesva om eigenlijk op layer te houden dat je op vl-layer een "vl-layer-style" zet (dus de breaking change) door te voeren, maar wel toe te laten om op vl-layer-style de "declaratief geconfigureerde" ol style te "overriden"?
Dan doen we een breaking change maar bieden we wel een makkelijk alternatief?

Reply:
Net even geprobeerd, maar dat is lastiger: een keer je de stijl gedefinieerd hebt, wordt die op de parent gezet, dus nadien nog de style gaan aanpassen, moet je ook nog de parent gaan beginnen updaten. Wat wel programmatorisch al mogelijk is, is zelf een subklasse maken en de getter van VlMapLayerStyle overriden en die toevoegen aan de VlMapLayer, zoals bij VlMapLayerCircleStyle eigenlijk al wordt gedaan. Zo wordt die style bij construction al juist gezet.

Reply:
Ja, dan zult ge met @driesva moeten overeenkomen omtrent welke aanpak OK is. Ik vermoed voor ons dat de puur declaratieve OK is.

Reply:
Na overleg met @Pieter-Beckers: het behouden van die initiële manier (OL style) of een breaking change met een alternatief (zoals zelf te maken subclass) is in orde wat mij betreft.